Model runs for phase 1b (#7)
ccmmf/workflows: phase_1b_ca_woody/validate.Rmd

  start = 2016,
  end = 2016,
  vars = "AbvGrndWood"
)

# totsoilcarb_23 <- read_ncdir(
#  file.path(outdir, "out"),
#  start.year = 2022,
#  end.year = 2023,
#  vars = "TotSoilCarb"
# )
```

## Cleanup: Remove sites missing LAI

A few sites were run with missing values for initial LAI.
The IC calculation logic did not handle these correctly and resulted in
these sites being simulated with a (too high) default starting biomass
instead of the provided LandTrendr values.

To identify these we could reparse IC files looking for missing values,
but that's slow. Instead I'll look for sites whose AbvGrndWood at start of
simulation in 2016 is within float error of 18 kgC/m2 (that is, a total woody
mass of 30 kgC/m2 with coarse and fine roots each set to be 20% of the total).

```{r badsites, cache = TRUE}
lai_missing <- abvgrdwood_16 |>
  ungroup() |>
  filter(posix == min(posix), AbvGrndWood > 17.5) |>
  distinct(site)
lai_missing

abvgrdwood_23 <- abvgrdwood_23 |>
  anti_join(lai_missing)
```

## Modeled biomass vs LandTrendr composite biomass

```{r extract-landtrendr, cache=TRUE}
point_buf <- site_info |>
  terra::vect(crs = "epsg:4326") |>
  terra::project(landtrendr_2023) |>

>   # TODO what is the right buffer distance to use here?

Author: Eventually yeah. In fact that's probably the better comment to leave
```suggestion
  # TODO load field polygons and clip to those
```

    mean_AGB_landtrendr = mean(AGB_median),
    mean_AGB_sipnet = mean(AbvGrndWood)
  )
bind_rows(agb_crop_summary, agb_tot_summary)

```

```{r agb-fit}
agb_fits <- agb_with_lt |>
  ungroup() |>
  nest_by(ens_num) |>
  mutate(
    fit = list(lm(AbvGrndWood ~ AGB_median, data = data)),
    r2 = summary(fit)$adj.r.squared,
    nse = 1 - (
      sum((data$AGB_median - data$AbvGrndWood)^2) /
        sum((data$AGB_median - mean(data$AGB_median))^2)
    ),
    rmse = sqrt(mean((data$AGB_median - data$AbvGrndWood)^2))
  )
agb_fits |> select(-data, -fit)
agb_fits |>
  ungroup() |>
  summarize(across(r2:rmse, c(mean = mean, sd = sd))) |>
  pivot_longer(everything(), names_to = c("stat", ".value"), names_sep = "_")
```

```{r plot-agb}
agb_with_lt |>
  ggplot() +
  aes(AGB_median, AbvGrndWood) +
  geom_abline(lty = "dashed") +
  geom_point() +
  geom_smooth(method = "lm") +
  facet_wrap(~ens_num) +
  labs(
    x = "LandTrendr 2023 median AGB, kg C/m2",
    y = "Sipnet AbvGrndWood 2023, kg C/m2")
```

## Modeled biomass vs allometry

Here we use a set of allometric equations for almonds, walnuts, pistachios,
and oranges developed by CARB as part of the 2018 NWL Inventory.

> Note that at least for this set of design points, no sites were specifically

Reviewer: ```suggestion
Note that at least for this set of design points, no sites were categorized as
```

fruits (D1, D3, D5, D16) or miscellaneous deciduous (D10) were excluded from the
allometric comparison.

```{r allom-fn}
allom_kgC_m2 <- function(age, species = c("almond", "walnut", "pistachio", "orange")) {
  if (any(age > 50)) stop("this allometry only valid to age 50")
  species <- match.arg(species)

  dens_m <- c(almond = -15.45, walnut = -19.97, pistachio = -24.7, orange = -34.71)
  dens_b <- c(almond = 125.72, walnut = 103.5, pistachio = 184.93, orange = 223.83)
  carbon_m <- c(almond = 12.9823, walnut = 9.3295, pistachio = 1.2879, orange = 2.4435)
  carbon_b <- c(almond = 1.3923, walnut = 1.6121, pistachio = 1.8835, orange = 1.3712)

  trees_acre <- dens_m[species] * log(age) + dens_b[species]
  lbs_tree <- carbon_m[species] * (age^carbon_b[species])

  PEcAn.utils::ud_convert(trees_acre * lbs_tree, "lb acre-1", "kg m-2")
}
```

```{r allom-read}
site_allom_C <- site_age |>
  mutate(species = case_when(
    crop_code == "C" ~ "orange",
    crop_code == "C3" ~ "orange",
    crop_code == "D12" ~ "almond",
    crop_code == "D13" ~ "walnut",
    crop_code == "D14" ~ "pistachio",
    TRUE ~ NA_character_
  )) |>

>   # TODO want to do anything other than drop sites with nontarget species?

Reviewer: Not sure what you have in mind, but I think that the existing comparison is straighforward, clear, sufficient.

Author: Fair, and the prose has more details on this too. Will delete the TODO
```suggestion
```
